src/content_cao.cpp

```diff
@@ -1302,14 +1308,17 @@ void GenericCAO::updateTexturePos()
 	}
 }
 
-void GenericCAO::updateTextures(const std::string &mod)
```

Comment:
doesn't work using the reference but I'd be glad if someone could explain to me why.

once the reference is in place previous and current are set to same value resulting in wrong behavior.